pkgdown-netlify-preview/pkgdown-netlify-preview.yaml

```
@@ -22,7 +28,7 @@ jobs:
       GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
       NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_AUTH_TOKEN }}
       NETLIFY_SITE_ID: ${{ secrets.NETLIFY_SITE_ID }}
-      isPush: ${{ github.event_name == 'push' || github.event_name == 'workflow_dispatch' }}
+      PUBLISH: ${{ github.event_name == 'push' || (github.event_name == 'workflow_dispatch' && inputs.publish) }}
```

Clarifying question about the expected behavior in the context of our release process....

When do we want to trigger a commit to the `gh-pages` branch? Only on release? When we push to the stable dev release branch (i.e., main)?

---

We want to trigger a commit to `gh-pages` branch whenever a push happens to the main branch and on release.

The majority of these pushes will go to the `dev/` folder because of the release process (see https://pkgdown.r-lib.org/reference/build_site.html?q=dev#development-mode).

Only on releases will the prod site be updated.

---

I've updated the initial comment to reflect what is published.

---

I didn't do a super deep dive into the GitHub docs, but after a bit of searching, it's not clear to me if a release would result in this `PUBLISH` variable being set to true.

Will this work as expected for a release?

---

🤔

---

I think the answer is "no" because `github.event_name` will be `"release"` for that event, which means that it's _never_ worked on release. It's also a bit moot because release events will generally follow push events 🤷🏽

---

This would fix things up

```suggestion
      PUBLISH: ${{ github.event_name == 'push' || github.event_name == 'release' || (github.event_name == 'workflow_dispatch' && inputs.publish) }}
```

---

> It's also a bit moot because release events will generally follow push events 🤷🏽

The word "generally" is doing a lot of work in this sentence. This, however does not follow for hotfix releases where the release comes _before_ the push event, which is exactly what happened in https://github.com/hubverse-org/hubVis/pull/49